cli/iconBuilder.js

```diff
@@ -0,0 +1,167 @@
+const fs = require('fs'),
+	path = require('path');
+
+const imagePath = path.join(__dirname, '../components/icons/images');
+const outputPath = path.join(__dirname, '../.generated/icons');
+
+async function getSvgsInDir(dir) {
+	return new Promise((resolve, reject) => {
+		const dirPath = path.join(imagePath, dir);
+		fs.stat(dirPath, (err, stat) => {
+			if (err) return reject(err);
+			if (!stat.isDirectory()) return resolve([]);
+			fs.readdir(dirPath, (err, files) => {
+				if (err) return reject(err);
+				const svgs = files.filter((file) => {
+					return (path.extname(file) === '.svg');
+				}).map((file) => {
+					return {category: dir, name: file.substr(0, file.length - 4)};
+				});
+				resolve(svgs);
+			});
+		});
+	});
+}
+
+async function getSvgs() {
```

Comment:
Seems like this function could be simplified to:
```
async function getSvgs() {
	let svgs = [];
	const files = fs.readdirSync(imagePath);
	files.forEach((file) => {
		const newSvgs = await getSvgsInDir(files[i]);
		svgs = svgs.concat(newSvgs);
	}
	return svgs;
}
```
A check for the directory could be added (`	fs.existsSync(...)`), but that directory is guaranteed to be there ya?

If you decide to stay with the async versions of the file system API, it seems like the `resolve` could be moved out of the loop.  Also if there are no files, would the returned Promise be left in a pending state?

Reply:
I tried moving the `resolve` out, and weirdly it runs immediately... like that `await` call doesn't block the loop. I was surprised.

Reply:
Wow... weird.